plugins/analysis-stempel/src/main/java/org/elasticsearch/index/analysis/pl/PolishStopTokenFilterFactory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.analysis.pl;
+
+
+import org.apache.lucene.analysis.CharArraySet;
+import org.apache.lucene.analysis.StopFilter;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.pl.PolishAnalyzer;
+import org.apache.lucene.search.suggest.analyzing.SuggestStopFilter;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.analysis.AbstractTokenFilterFactory;
+import org.elasticsearch.index.analysis.Analysis;
+
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonMap;
+
+public class PolishStopTokenFilterFactory extends AbstractTokenFilterFactory {
```

Comment:
It seems weird to have a `PolishStopTokenFilterFactory` where the list of stopwords can be overridden in a setting (`stopwords`). Did you add this for the documentation example ? Can we instead use a `stop` filter and set the polish list of stopwords explicitly ?

Reply:
The stopword list is in th plugin jar, I tried a few things loading it via the settings that didn't work (one of them the obvious thing trying to load the polish stopwords with `"stopwords" : "_polish_"`  but for that to work I think the stopworks would need to get registered in the Analysis#NAMED_STOP_WORDS map, but we don't seem to have a straight forward way to do that from a plugin). Accessing the "stopwords.txt" file via "stopwards_path" in the plugin jar didn't work either, it tries to load files from the "conf" directory. 
I can change the "polish_stop" to not be configurable, but this is the same we do e.g. in `JapaneseStopTokenFilterFactory` where I got the idea from.

Reply:
Ok thanks for explaining. However if we add this filter we should have a documentation similar to https://www.elastic.co/guide/en/elasticsearch/plugins/master/analysis-kuromoji-stop.html. I think it's ok if we allow custom list of stopwords in this filter but we need to ensure (I think it's already the case) that `_polish_` can be used to add the default list. It won't work in the default `stop` filter but this shouldn't be an issue.